### PR TITLE
Add @types/next-server / port @types/next lib

### PR DESCRIPTION
**Pull request by ajliv**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by ajliv** [comment]
[Next.js](https://github.com/zeit/next.js) is splitting it's lib into a new next-server package. Currently on the [canary](https://github.com/zeit/next.js/tree/canary) branch awaiting merge into master but published to npm as `next-server`

**Comment by sheetalkamat** [comment]
Since these definitions were part of next, adding the code reviewers from there.
cc: @Dru89 @brikou @jthegedus  @resir014 @scottdj92 @joaovieira for review

File: types/next-server/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
+// Type definitions for next-server 7.0
+// Project: https://github.com/zeit/next.js/packages/next-server
+// Definitions by: AJ Livingston <https://github.com/ajliv>
```
Comment (resir014) [inline types/next-server/index.d.ts]: Please don't forget to add the other maintainers from `next` here. And add your name there as well.

File: types/next/config.d.ts [hunk 2]
```diff
@@ -1 +1,4 @@
-export default function(): { serverRuntimeConfig: any; publicRuntimeConfig: any };
+import config from 'next-server/config';
+
+export * from 'next-server/config';
+export default config;
```
Comment (joaovieira) [inline types/next/config.d.ts]: or 

```ts
export * from 'next-server/config';
export { default } from 'next-server/config';
```

**Review by joaovieira — CHANGES_REQUESTED** [review]
Tiny thing. I think we should keep the new package clean without bringing the deprecations over. I feel there's also duplicate tests now?

**Comment by ajliv** [comment]
@joaovieira, There are duplicate tests between the packages. I left them to ensure there were no breaking changes to the `next` package since it's just re-exporting from `next-server` and the functionality remains the same.

**Comment by joaovieira** [comment]
Yeah, I feel they are useful too _for now_. Until the packages are decoupled and mutually exclusive 👍

**Review by joaovieira — APPROVED** [review]

**Pull request merged**